src/scheduler/server_info.c

```diff
@@ -268,6 +270,16 @@ query_server(status *pol, int pbs_sd)
 		return NULL;
 	}
 
+	if (sinfo->nodes[0] == NULL) {
+		snprintf(log_buffer, sizeof(log_buffer), "No nodes found, skipping this cycle");
+		schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_SERVER, LOG_ERR, __func__, log_buffer);
+		pbs_statfree(server);
+		sinfo->fairshare = NULL;
+		free(sinfo->nodes);
+		free_server(sinfo, 0);
+		return NULL;
+	}
```

Comment:
Is query_server() the right place for this check, or is query_nodes()?  If query_nodes() returns NULL, we bail out of the cycle.

Reply:
I think I agree with Bhroam, query_nodes sounds like a better place for this check